### PR TITLE
PTV-1786: Add a small workspace delegation helper

### PR DESCRIPTION
**Pull request by MrCreosote**
Hides some of the annoyances of the workspace client API.

Also some minor cleanup

### CODE REVIEW

File: src/us/kbase/workspace/kbase/LocalTypeServerMethods.java [hunk 1]
```diff
@@ -114,12 +115,12 @@ public Map<String,String> registerTypespec(
 		final Map<TypeDefName, TypeChange> res;
 		if (params.getMod() != null) {
 			 res = types.compileTypeSpec(
-					getUser(authPart), params.getMod(),
+					getUser(token), params.getMod(),
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/kbase/LocalTypeServerMethods.java:118]: will you add a test for this?

**Review by ialarmedalien — APPROVED** [review]
Looks fine

File: src/us/kbase/workspace/kbase/LocalTypeServerMethods.java (hunk 1 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/LocalTypeServerMethods.java:118]: Not right now. This is the same code we talked about before, where I just moved it from the server class into this class verbatim so I could write an alternate implementation.

Comment (ialarmedalien) [reply src/us/kbase/workspace/kbase/LocalTypeServerMethods.java:118]: Got it, cool.

**Comment by MrCreosote** [comment]
The difference between 42e6d52 and d9bcf5c is just merge commits so the code coverage is accurate. Not sure why codecov didn't pick up the new builds.

**Pull request merged**